fcsparser/api.py

```diff
@@ -206,16 +207,23 @@ def _extract_text_dict(self, raw_text):
                 raise ParserFeatureNotImplementedError(msg)
 
         # Below 1:-1 used to remove first and last characters which should be reserved for delimiter
-        # Also, the delimiter is "quoted" by being repeated (two consecutive delimiters)
-        delimiter_split_arr = [x.split(delimiter) for x in raw_text[1:-1].split(delimiter * 2)]
-
-        raw_text_segments = delimiter_split_arr[0]
-        for partial_segment in delimiter_split_arr[1:]:
-            raw_text_segments[-1] += (delimiter + partial_segment[0])
-            raw_text_segments.extend(partial_segment[1:])
+        raw_text = raw_text[1:-1]
+
+        ed = re.escape(delimiter)  # escaped for regex, not FCS format
+        # The delimiter is escaped by being repeated (two consecutive delimiters). This regex finds
+        # odd-length repetitions of the delimiter. Since each pair is escaped, only the last
+        # delimiter in odd-length repetitions truly delimits TEXT keys and values.
+        regex = "(?<!" + ed + ")(" + ed * 2 + ")*" + ed + "(?!" + ed + ")"
```

Comment:
Since we're allowing the test to construct the regular expression dynamically, we should check for a few edge cases with the delimiter being `\` rather than `/` 

You may nee to specify that the regexp is a raw string `regex = r'...'`

Reply:
At least one of the test FCS files has `\` as a delimiter. I had to add the `re.escape()` to handle that.